apps/client/app/components/PurchaseOrderForm.tsx

```diff
@@ -0,0 +1,167 @@
+        <label className="block text-accent-content">Expected Delivery Date</label>
+        <Controller
+          name="expected_delivery_date"
+          control={control}
+          render={({ field }) => <input type="date" {...field} className="mt-1 block w-full border border-accent rounded-md shadow-sm" />}
+        />
+        {errors.expected_delivery_date && <p className="text-red-500 text-xs">{errors.expected_delivery_date.message}</p>}
+      </div>
+      <div className="mb-4">
+        <h3 className="text-lg font-semibold text-accent-content">Line Items</h3>
+        {fields.map((field, index) => (
+          <div key={field.id} className="mb-2">
+            <label className="block text-accent-content">Item</label>
+            <Controller
+              name={`purchase_order_line_items.${index}.item_id`}
+              control={control}
+              defaultValue={field.item_id}
+              render={({ field: selectField }) => (
+                <select {...selectField} className="mt-1 block w-full border border-accent rounded-md shadow-sm">
+                  {items.map((item) => (
+                    <option key={item.id} value={item.id}>{item.name}</option>
+                  ))}
+                </select>
+              )}
+            />
+            {errors.purchase_order_line_items?.[index]?.item_id && <p className="text-red-500 text-xs">{errors.purchase_order_line_items?.[index]?.item_id?.message}</p>}
+            <label className="block text-accent-content">Quantity</label>
+            <Controller
+              name={`purchase_order_line_items.${index}.quantity`}
+              control={control}
+              render={({ field }) => <input type="number" {...field} className="mt-1 block w-full border border-accent rounded-md shadow-sm" />}
+            />
+            {errors.purchase_order_line_items?.[index]?.quantity && <p className="text-red-500 text-xs">{errors.purchase_order_line_items?.[index]?.quantity?.message}</p>}
+            <label className="block text-accent-content">Unit Cost</label>
+            <Controller
+              name={`purchase_order_line_items.${index}.unit_cost`}
+              control={control}
+              render={({ field }) => <input type="number" {...field} className="mt-1 block w-full border border-accent rounded-md shadow-sm" />}
+            />
+            {errors.purchase_order_line_items?.[index]?.unit_cost && <p className="text-red-500 text-xs">{errors.purchase_order_line_items?.[index]?.unit_cost?.message}</p>}
+            <button type="button" onClick={() => remove(index)} className="mt-2 text-red-500">Remove Line Item</button>
+          </div>
+        ))}
+        <button type="button" onClick={() => append({ id: 0, item_id: 0, quantity: 0, unit_cost: 0 })} className="mt-4 text-blue-500">Add Line Item</button>
+      </div>
+      <button type="submit" className="w-full py-2 px-4 bg-blue-500 text-white rounded-md shadow-sm">Submit</button>
+    </form>
+  );
```

Comment:
For the navigation, I wanted a full page reload so that the purchase order list gets updated from the server. 

I will implement ensure I have the necessary error handling in the APIs. 

In terms of adding the confirmation dialog before removing line items, I didn't do that because you need to click submit before anything is deleted. I certainly can if the team decides that is the desired UX.